src/notebook.rs

```diff
@@ -38,7 +38,11 @@ impl NoteBook {
         };
 
         let notebook_clone = self.notebook.clone();
-        button.connect_clicked(move |_| notebook_clone.remove_page(index as i32));
+        let widget_clone = widget.clone();
```

Comment:
If you don't need `notebook` and `widget` after this, you can just move them into the closure without cloning.